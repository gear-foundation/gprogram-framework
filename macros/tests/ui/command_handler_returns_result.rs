use gprogram_framework_macros::{command_handlers, gprogram};

#[gprogram]
mod gprogram {
    use super::*;

    command_handlers!(
        fn do_this(value: u32) -> u32 {
            value
        }
    );
}

fn main() {}
