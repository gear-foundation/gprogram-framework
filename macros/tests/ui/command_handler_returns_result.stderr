error[E0308]: mismatched types
  --> tests/ui/command_handler_returns_result.rs:7:5
   |
7  | /     command_handlers!(
8  | |         fn do_this(value: u32) -> u32 {
9  | |             value
10 | |         }
11 | |     );
   | |     ^
   | |     |
   | |_____expected `Result<_, _>`, found `u32`
   |       expected due to this
   |
   = note: expected enum `Result<_, _>`
              found type `u32`
   = note: this error originates in the macro `command_handlers` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try wrapping the expression in a variant of `Result`
   |
7  ~     gprogram::commands_handlers_wasm::gstd::Ok(command_handlers!(
8  |         fn do_this(value: u32) -> u32 {
9  |             value
10 |         }
11 ~     ));
   |
7  ~     gprogram::commands_handlers_wasm::gstd::Err(command_handlers!(
8  |         fn do_this(value: u32) -> u32 {
9  |             value
10 |         }
11 ~     ));
   |

error[E0308]: mismatched types
  --> tests/ui/command_handler_returns_result.rs:7:5
   |
7  | /     command_handlers!(
8  | |         fn do_this(value: u32) -> u32 {
9  | |             value
10 | |         }
11 | |     );
   | |     ^
   | |     |
   | |_____expected `u32`, found `Result<_, _>`
   |       arguments to this enum variant are incorrect
   |
   = note: expected type `u32`
              found enum `Result<_, _>`
note: tuple variant defined here
  --> tests/ui/command_handler_returns_result.rs:7:5
   |
7  | /     command_handlers!(
8  | |         fn do_this(value: u32) -> u32 {
9  | |             value
10 | |         }
11 | |     );
   | |_____^
   = note: this error originates in the macro `command_handlers` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `encode` found for enum `gprogram::CommandResponses` in the current scope
  --> tests/ui/command_handler_returns_result.rs:7:5
   |
7  |       command_handlers!(
   |  _____^
8  | |         fn do_this(value: u32) -> u32 {
9  | |             value
10 | |         }
11 | |     );
   | |     ^
   | |     |
   | |_____method `encode` not found for this enum
   |       method not found in `CommandResponses`
   |
  ::: $CARGO/parity-scale-codec-3.6.5/src/codec.rs
   |
   |       fn encode(&self) -> Vec<u8> {
   |          ------ the method is available for `gprogram::CommandResponses` here
   |
   = help: items from traits can only be used if the trait is in scope
   = note: this error originates in the macro `command_handlers` (in Nightly builds, run with -Z macro-backtrace for more info)
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
7  |     use crate::gprogram::commands_scale_codec::Encode;
   |
